Tests/NIOHTTPCompressionTests/HTTPResponseCompressorTest.swift

```diff
@@ -688,6 +724,234 @@ class HTTPResponseCompressorTest: XCTestCase {
             }
         }
     }
+    
+    func testConditionalCompressionEnabled() throws {
+        let predicateWasCalled = expectation(description: "Predicate was called")
+        let compressor = HTTPResponseCompressor { responseHeaders, isCompressionSupported in
+            predicateWasCalled.fulfill()
```

Comment:
We should probably fulfil the predicate last.

Reply:
Put them inside a defer instead so this is guaranteed, unless that would lead to a bad outcome? Its only really trying to make sure the predicate isn't silently skipped.

Reply:
Defer is just fine.